services/types.go

```diff
@@ -1,7 +1,24 @@
 package services
 
+type OnboardingSubjectRole string
+
+type OnboardingSubjectRoleSpec struct {
+	Role          OnboardingSubjectRole `json:"role"`
```

Comment:
you are trying to make `role` as mandatory but the original could be empty quota, thought of having `OnboardingSubjectRoleSpec` as a pointer in parent struct?

Reply:
`original could be empty quota` Could you elaborate on what you mean by original could be empty quota?